app/controllers/api/v2/app_groups_controller.rb

```diff
@@ -179,25 +184,35 @@ def deactivated_by_cluster_name
         code: 400,
       }, status: :bad_request) && return
     end
-  
-    @helm_infrastructure = HelmInfrastructure.joins(:app_group).find_by(cluster_name: cluster_name, app_groups: { name: app_group_name })
-  
-    if @helm_infrastructure.blank? || !@helm_infrastructure.active?
+
+    app_group = AppGroup.find_by(cluster_name: cluster_name, name: app_group_name, status: :ACTIVE)
+    # if not found
+    if app_group.blank?
       render(json: {
         success: false,
-        errors: ['Helm Infrastructure not found'],
+        errors: ['App Group not found'],
         code: 404,
       }, status: :not_found) && return
     end
-    
-    app_group = @helm_infrastructure.app_group
-    barito_apps = app_group.barito_apps 
-    barito_apps.each do |app|
+
+    # set the appgroup to INACTIVE
+    app_group.update!(status: :INACTIVE)
+
+    # set each app as INACTIVE
+    app_group.barito_apps.each do |app|
       app.update_status('INACTIVE') if app.status == BaritoApp.statuses[:active]
     end
-  
-    @helm_infrastructure.update_provisioning_status('DELETE_STARTED')
-    DeleteHelmInfrastructureWorker.perform_async(@helm_infrastructure.id)
+
+    if Figaro.env.ARGOCD_ENABLED == 'true'
+      app_group.helm_infrastructures.each do |helm_infrastructure|
+        ArgoDeleteWorker.perform_async(helm_infrastructure.id)
+      end
+    else
+      app_group.helm_infrastructures.each do |helm_infrastructure|
+        helm_infrastructure.update_provisioning_status('DELETE_STARTED')
+        DeleteHelmInfrastructureWorker.perform_async(helm_infrastructure.id)
+      end
+    end
```

Comment:
is deactivating means removing the whole kubernetes resources? isn't only mark this resource as inactive but still keep the kubernetes resource?

Reply:
not sure, mas @fadlinurhasan might be able to answer this

Reply:
is deactivating means removing the whole kubernetes resources? isn't only mark this resource as inactive but still keep the kubernetes resource?

Yes Jun, deactivating app group will remove all k8s resource and this deactivating only can be done by barito-devs. which I can see it already there on the policy.